rest_framework/fields.py

```diff
@@ -1042,10 +1042,16 @@ def to_internal_value(self, data):
         except KeyError:
             self.fail('invalid_choice', input=data)
 
+    def representation_value(self, value):
+        try:
+            return self.choice_strings_to_values[six.text_type(value)]
+        except KeyError:
+            return value
```

Comment:
Actually, perhaps a `self.choice_strings_to_values(six.text_type(value), value)` is better here?